apollo-federation/src/merge.rs

```diff
@@ -591,6 +598,25 @@ impl Merger {
             }
         }
     }
+
+    fn merge_scalar_type(
+        &self,
+        types: &mut IndexMap<Name, ExtendedType>,
+        subgraph_name: Name,
+        scalar_name: NamedType,
+        ty: &Node<ScalarType>,
+    ) {
+        let existing_type = types
+            .entry(scalar_name.clone())
+            .or_insert(copy_scalar_type(scalar_name, ty));
+        if let ExtendedType::Scalar(e) = existing_type {
+            let join_type_directives =
+                join_type_applied_directive(subgraph_name.clone(), iter::empty(), false);
+            e.make_mut().directives.extend(join_type_directives);
+        } else {
+            // conflict?
```

Comment:
Should we change the signature of this method to return a `Result` and have it bubble up? Might help us catch issues early on if we encounter areas where this conflict happens.

Reply:
this was a copy-paste from a similar method. i'll add a subtask to the merge ticket to return to the conflict handling